drivers/display/display_sh1122.c

```diff
@@ -0,0 +1,515 @@
+}
+
+static inline int sh1122_write_bus_cmd_i2c(const struct device *dev, const uint8_t cmd,
+					   const uint8_t *data, size_t len)
+{
+	const struct sh1122_config *config = dev->config;
+	static uint8_t buf[SH1122_MAXIMUM_CMD_LENGTH];
+
+	if (len > SH1122_MAXIMUM_CMD_LENGTH - 1) {
+		return -EINVAL;
+	}
+
+	buf[0] = cmd;
+	memcpy(&(buf[1]), data, len);
+
+	return i2c_burst_write_dt(&config->i2c, SH1122_CONTROL_ALL_BYTES_CMD, buf, len + 1);
+}
+
+static inline int sh1122_set_hardware_config(const struct device *dev)
+{
+	const struct sh1122_config *config = dev->config;
+	int err;
+
+	err = config->write_cmd(dev, SH1122_SET_DISPLAY_START_LINE(config->start_line), NULL, 0);
+	if (err < 0) {
+		return err;
+	}
+	err = config->write_cmd(dev, SH1122_SET_DISPLAY_OFFSET, &config->display_offset, 1);
+	if (err < 0) {
+		return err;
+	}
+
+	err = config->write_cmd(dev, SH1122_SET_DISPLAY_RAM, NULL, 0);
+	if (err < 0) {
+		return err;
+	}
+	err = config->write_cmd(dev, SH1122_SET_NORMAL_DISPLAY, NULL, 0);
+	if (err < 0) {
+		return err;
+	}
+	err = config->write_cmd(
+		dev, config->inv_com ? SH1122_SET_COM_ORDER_01 : SH1122_SET_COM_ORDER_10, NULL, 0);
+	if (err < 0) {
+		return err;
+	}
+	err = config->write_cmd(
+		dev, config->inv_seg ? SH1122_SET_SEG_ORDER_01 : SH1122_SET_SEG_ORDER_10, NULL, 0);
+	if (err < 0) {
+		return err;
+	}
+	err = config->write_cmd(dev, SH1122_SET_VSL(config->low_voltage), NULL, 0);
+	if (err < 0) {
+		return err;
+	}
+	err = config->write_cmd(dev, SH1122_SET_MULTIPLEX_RATIO, &config->multiplex_ratio, 1);
+	if (err < 0) {
+		return err;
+	}
+
+	err = config->write_cmd(dev, SH1122_SET_PHASE_LENGTH, &config->phase_length, 1);
+	if (err < 0) {
+		return err;
+	}
+	err = config->write_cmd(dev, SH1122_SET_OSC_FREQ, &config->oscillator_freq, 1);
+	if (err < 0) {
+		return err;
+	}
+	err = config->write_cmd(dev, SH1122_SET_VSEGM, &config->precharge_voltage, 1);
+	if (err < 0) {
+		return err;
+	}
+	err = config->write_cmd(dev, SH1122_SET_VCOMH, &config->vcomh_voltage, 1);
+	if (err < 0) {
+		return err;
+	}
+	return config->write_cmd(dev, SH1122_SET_DCDC, &config->dc_dc, 1);
+}
+
+static int sh1122_resume(const struct device *dev)
+{
+	const struct sh1122_config *config = dev->config;
+	int err;
+
+	err = config->write_cmd(dev, SH1122_DISPLAY_ON, NULL, 0);
+	if (err < 0) {
+		return err;
+	}
+	return config->release_bus(dev);
+}
+
+static int sh1122_suspend(const struct device *dev)
+{
+	const struct sh1122_config *config = dev->config;
+	int err;
+
+	err = config->write_cmd(dev, SH1122_DISPLAY_OFF, NULL, 0);
+	if (err < 0) {
+		return err;
+	}
+	return config->release_bus(dev);
+}
+
+#if DT_HAS_COMPAT_ON_BUS_STATUS_OKAY(sinowealth_sh1122, mipi_dbi)
+static int sh1122_write_pixels_mipi(const struct device *dev, const uint8_t *buf,
+				    const struct display_buffer_descriptor *desc)
+{
+	const struct sh1122_config *config = dev->config;
+	struct display_buffer_descriptor mipi_desc;
+
+	mipi_desc.buf_size = desc->width / 2;
```

Comment:
It could be nice to have "4-bit per pixel" or "2 pixel per byte" appear at least once in the driver. Maybe it is  be obvious for people experimented with displays though.

Reply:
`/* Convert one line to pixelx+1 (3:0) and pixelx (7:4) */`  on line 300 ?

Reply:
Either way, only a suggestion, so resolving the thread :)